fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexScrubbingTest.java

```diff
@@ -0,0 +1,302 @@
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.SIMPLE_TEXT_SUFFIXES_WITH_PRIMARY_KEY_SEGMENT_INDEX;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.createComplexDocument;
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.createSimpleDocument;
+import static com.apple.foundationdb.record.lucene.directory.InjectedFailureRepository.Flags.LUCENE_MAINTAINER_SKIP_INDEX_UPDATE;
+import static com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexTestUtils.SIMPLE_DOC;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class LuceneIndexScrubbingTest extends FDBLuceneTestBase {
+
+    private TestingIndexMaintainerRegistry registry;
+    private boolean flipBoolean = false;
+
+    @BeforeEach
+    public void beforeEach() {
+        registry = new TestingIndexMaintainerRegistry();
+    }
+
+    private void rebuildIndexMetaData(final FDBRecordContext context, final String document, final Index index) {
+        Pair<FDBRecordStore, QueryPlanner> pair = LuceneIndexTestUtils.rebuildIndexMetaData(context, path, document, index, isUseCascadesPlanner());
+        this.recordStore = pair.getLeft();
+        this.planner = pair.getRight();
+    }
+
+    private static Stream<Arguments> threeBooleanArgs() {
+        return Stream.of(false, true)
+                .flatMap(isSynthetic -> Stream.of(false, true)
+                        .flatMap(isPartitioned -> Stream.of(false, true)
+                                .map(isGrouped -> Arguments.of(isSynthetic, isGrouped, isPartitioned))));
+    }
+
+    private boolean yinYang() {
+        flipBoolean = !flipBoolean;
+        return flipBoolean;
+    }
+
+    @Nonnull
+    protected FDBRecordStore.Builder getStoreBuilderWithRegistry(@Nonnull FDBRecordContext context,
+                                                                 @Nonnull RecordMetaDataProvider metaData,
+                                                                 @Nonnull final KeySpacePath path) {
+        return super.getStoreBuilder(context, metaData, path).setIndexMaintainerRegistry(registry);
+    }
+
+    @ParameterizedTest
+    @MethodSource("threeBooleanArgs")
+    void luceneIndexScrubMissingDataModelNoIssues(boolean isSynthetic, boolean isGrouped, boolean isPartitioned) {
+        // Scrub a valid index, expect zero issues
+        final long seed = 7L;
+
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilderWithRegistry, pathManager)
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPartitionHighWatermark(isPartitioned ? 5 : 0)
+                .build();
+
+        for (int i = 0; i < 14; i++) {
+            dataModel.setReverseSaveOrder(yinYang());
```

Comment:
Nit: Also, given the use case, you could do the following, and not create additional state:
```suggestion
            dataModel.setReverseSaveOrder(i % 2 == 0);
```